File: 0050-powx-n/0050-powx-n.java
class Solution {
    public double myPow(double x, int n) {
        if(n == 0){
            return 1.00;
        }

        double result = myPow(x, n/2);

        if(n % 2 == 0){
            return result * result;
        }else {
            if(n < 0){
                return result * result * (1/x);
            }
            else{
                return result * result * x;
            }
        }
    }
}

File: 0050-powx-n/0050-powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0:
            return 1.0
        
        result = self.myPow(x, int(n / 2))

        if n % 2 == 0:
            return result * result
        else:
            if n > 0:
                return result * result * x
            else:
                return result * result * (1/x)
            
        
        return result